_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+
+## Turn the Angular application into a micro-frontend remote
+
+We want to use the schematic from `@angular-architects/module-federation` to turn the `mfe-profile` application into a micro-frontend and add the necessary configuration. We'll use port 4202 for this application. Add the schematic by running the following command in the terminal:
+
+```shell
+ng add @angular-architects/module-federation --project mfe-profile --port 4202
+```
+
+This schematic does the following:
+1. Updates the project's `angular.json` config file to add the port for the application and updates the builder to use a custom Webpack builder
+2. Creates the `webpack.config.js` files and scaffolds out default configuration for Module Federation
+
+First, let's add the new micro-frontend to the `shell` application by updating the configuration in `projects/mfe-profile/webpack.config.js`. In the middle of the file, there's a property for `plugins` with commented-out code. We need to finish configuring that. Since this application is a remote, we'll update the snippet of code under the comment:
+
+```js
+// For remotes (please adjust)
+```
+
+The defaults are mostly correct, except we have a module, not a component that we want to expose. If you want to expose a component instead, all you'd need to do is update which component to expose. Update the configuration snippet to expose the `ProfileModule` by matching the following code snippet:
+
+```js
+// For remotes (please adjust)
+name: "mfeProfile",
+filename: "remoteEntry.js",
+exposes: {
+  './Module': './projects/mfe-profile/src/app/profile/profile.module.ts',
+},
+```
+
+Now we can incorporate the micro-frontend in the `shell` application. Open `projects/shell/webpack.config.js`. Here is where you'll add the new micro-frontend so that the `shell` application knows how to access it. In the middle of the file, inside the `plugins` array, there's a property for `remotes`. The micro-frontend in the starter code, `mfeBasket`, is already added to the `remotes` object. There is where you'll add the remote for `mfeProfile` following the same pattern but replacing the port to 4202. Update your configuration to look like this.
+
+```js
+// For hosts (please adjust)
+remotes: {
+  "mfeBasket": "http://localhost:4201/remoteEntry.js",
+  "mfeProfile": "http://localhost:4202/remoteEntry.js"
+},
+```
+
+We can add update the code to incorporate the profile's micro-frontend. Open `projects/shell/src/app/app-routing.module.ts`. Add a path to the profile micro-frontend to the routes array using the path 'profile'. Your routes array should look like this.
```

Comment:
```suggestion
We can update the code to incorporate the profile's micro frontend. Open `projects/shell/src/app/app-routing.module.ts`. Add a path to the profile micro frontend to the routes array using the path 'profile'. Your routes array should look like this.
```
removed "add"
"Add a path to the profile micro frontend to the routes array" - is this as intended? feels like there might be a typo?

Reply:
Changed to 
> We can update the code to incorporate the profile's micro frontend. Open `projects/shell/src/app/app-routing.module.ts`. Add a path to the profile micro frontend in the routes array using the path 'profile'. Your routes array should look like this.

Reply:
wfm! Thank you.